src/cmd/wallet.rs

```diff
@@ -270,35 +272,73 @@ impl SpendOpt {
             (psbt, vec![])
         };
 
-        party.wallet().sign(&mut psbt, SignOptions::default())?;
-
-        let finalized = party
-            .wallet()
-            .finalize_psbt(&mut psbt, SignOptions::default())?;
-
-        assert!(finalized, "transaction must be finalized at this point");
-
-        let (output, txid) = cmd::decide_to_broadcast(
-            party.wallet().network(),
-            party.wallet().client(),
-            psbt,
-            yes,
-            print_tx,
-        )?;
-
-        if let Some(txid) = txid {
-            if !print_tx {
-                for bet_id in claiming_bet_ids {
-                    if let Err(e) = party.take_next_action(bet_id, false) {
-                        eprintln!(
-                            "error updating state of bet {} after broadcasting claim tx {}: {}",
-                            bet_id, txid, e
-                        );
+        let config = load_config(wallet_dir).context("loading config")?;
+
+        let output = match config.wallet_key {
+            WalletKey::SeedWordsFile {} => {
+                party.wallet().sign(&mut psbt, SignOptions::default())?;
+                let finalized = party
+                    .wallet()
+                    .finalize_psbt(&mut psbt, SignOptions::default())?;
+
+                let (output, txid) = cmd::decide_to_broadcast(
+                    party.wallet().network(),
+                    party.wallet().client(),
+                    psbt,
+                    yes,
+                    print_tx,
+                )?;
+
+                if let Some(txid) = txid {
```

Comment:
I think `decide_to_broadcast` and this updating state should go after the match and they are done regardless of the `wallet_key`. The match arms should just return a signed psbt.

(but this comment doesn't matter if we do the below idea)